examples/mixed_signal/tests/test_rescap.py

```diff
@@ -0,0 +1,168 @@
+        color_volt = "tab:red"
+        color_curr = "tab:blue"
+        ax_volt.set_title("rescap")
+        ax_volt.set_xlabel("Time (ns)")
+        ax_volt.set_ylabel("Voltage (V)", color=color_volt)
+        ax_curr = ax_volt.twinx()  # instantiate a second axis that shares the same x-axis
+        ax_curr.set_ylabel("Current (mA)", color=color_curr)  # we already handled the x-label with ax_volt
+
+        for node in nodes:
+            time, voltage, current = zip(*(datasets[node]))
+            if node.endswith("vout"):
+                alpha = 1.0
+            else:
+                alpha = 0.333
+            ax_volt.plot(time, voltage, color=color_volt, alpha=alpha,
+                         marker=".", markerfacecolor="black", linewidth=1, label="V({})".format(node))
+            ax_curr.plot(time, current, color=color_curr, alpha=alpha,
+                         marker=".", markerfacecolor="black", linewidth=1, label="I({})".format(node))
+
+        ax_volt.tick_params(axis="y", labelcolor=color_volt)
+        ax_curr.tick_params(axis="y", labelcolor=color_curr)
+        ax_volt.axhline(linestyle=":", color="gray")
+
+        mpl_align_yaxis(ax_volt, 0, ax_curr, 0)
+        fig.tight_layout()  # otherwise the right y-label is slightly clipped
+        fig.set_size_inches(11, 6)
+        fig.legend(loc="upper right", bbox_to_anchor=(0.8, 0.9), frameon=False)
+
+        self.tb_hdl._log.info("Writing file {}".format(graphfile))
+        fig.savefig(graphfile)
+
+
+async def run_test(tb_hdl):
+    """Run test for mixed signal resistor/capacitor simulation."""
+
+    tb_py = ResCap_TB(tb_hdl)
+
+    nodes_to_probe = ["tb_rescap.i_rescap.vdd", "tb_rescap.i_rescap.vout"]
+    # nodes_to_probe = ["tb_rescap.i_rescap.vdd", "tb_rescap.i_rescap.i_capacitor.p"]
+
+    probedata = defaultdict(list)
+
+    # dummy reads appear to be necessary for the analog solver
+    vdd = 0.0
+    tb_py.tb_hdl.vdd_val <= vdd
+    tb_py.tb_hdl.vss_val <= 0.0
+    tb_py.tb_hdl._log.info("Setting vdd={:.4} V".format(vdd))
+
+    data = await tb_py.get_sample_data(num=2, delay_ns=5, nodes=nodes_to_probe)
+    for node in nodes_to_probe:
+        probedata[node].extend(data[node])
+
+    vdd = 5.55
+    tb_py.tb_hdl.vdd_val <= vdd
+    tb_py.tb_hdl.vss_val <= 0.0
+    tb_py.tb_hdl._log.info("Setting vdd={:.4} V".format(vdd))
+
+    data = await tb_py.get_sample_data(num=60, delay_ns=5, nodes=nodes_to_probe)
+    for node in nodes_to_probe:
+        probedata[node].extend(data[node])
+
+    vdd = -3.33
+    tb_py.tb_hdl.vdd_val <= vdd
+    tb_py.tb_hdl.vss_val <= 0.0
+    tb_py.tb_hdl._log.info("Setting vdd={:.4} V".format(vdd))
+
+    data = await tb_py.get_sample_data(num=60, delay_ns=5, nodes=nodes_to_probe)
+    for node in nodes_to_probe:
+        probedata[node].extend(data[node])
+
+    tb_py.plot_data(datasets=probedata, nodes=nodes_to_probe, graphfile="rescap.png")
+
+
+# register the test
+factory = TestFactory(run_test)
+factory.generate_tests()
```

Comment:
I'd recommend not using `TestFactory` unless you are using the functionality it provides. Just decorate `run_tests` with `cocotb.test` Unless, you want to perhaps have a test for both `tb_rescap.i_rescap.vout` and `tb_rescap.i_rescap.i_capacitor.p`?

Reply:
``TestFactory`` was kind of meant to set up the user for extending this into a big test, but agreed, that's just distracting here.